File: frontend/src/lib/api/types/spatial-risk.ts
export interface SpatialCell {
  cell_id: string;
  center_lat: number;
  center_lon: number;
  hazard_type: string;
  record_type: string;
  average_risk: number;
  max_risk: number;
  min_risk: number;
  latest_risk: number;
  latest_assessment_time: string;
  assessment_count: number;
  high_risk_count: number;
  critical_risk_count: number;
  risk_category: string;
  provenance: string[];
}

export interface AggregationMetadata {
  time_window: string;
  hazard?: string;
  record_type: string;
  cell_size: number;
  total_cells: number;
  total_assessments: number;
}

export interface SpatialAggregationResponse {
  aggregation_metadata: AggregationMetadata;
  cells: SpatialCell[];
}

export interface ClusterMetadata {
  cluster_id: string;
  hazard_type: string;
  record_type: string;
  centroid_lat: number;
  centroid_lon: number;
  member_count: number;
  first_timestamp: string;
  latest_timestamp: string;
  duration_hours: number;
  average_risk: number;
  max_risk: number;
  high_risk_count: number;
  critical_risk_count: number;
  provenance: string[];
  quality_indicator: string;
}

export interface ClusterResponseMetadata {
  time_window: string;
  hazard?: string;
  record_type: string;
  total_clusters: number;
  total_assessments: number;
  spatial_threshold_km: number;
  temporal_threshold_hours: number;
}

export interface ClusterResponse {
  metadata: ClusterResponseMetadata;
  clusters: ClusterMetadata[];
}

File: backend/app/schemas/spatial_risk.py
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel

class SpatialCell(BaseModel):
    cell_id: str
    center_lat: float
    center_lon: float
    hazard_type: str
    record_type: str
    average_risk: float
    max_risk: float
    min_risk: float
    latest_risk: float
    latest_assessment_time: datetime
    assessment_count: int
    high_risk_count: int
    critical_risk_count: int
    risk_category: str
    provenance: List[str]

class AggregationMetadata(BaseModel):
    time_window: str
    hazard: Optional[str]
    record_type: str
    cell_size: float
    total_cells: int
    total_assessments: int

class SpatialAggregationResponse(BaseModel):
    aggregation_metadata: AggregationMetadata
    cells: List[SpatialCell]
